exo-go/src/app_setup/index_test.go

```diff
@@ -0,0 +1,155 @@
+package appSetup_test
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/Originate/exosphere/exo-go/src/app_config_helpers"
+	"github.com/Originate/exosphere/exo-go/src/app_setup"
+	"github.com/Originate/exosphere/exo-go/src/docker_helpers"
+	"github.com/Originate/exosphere/exo-go/src/logger"
+	"github.com/Originate/exosphere/exo-go/src/os_helpers"
+	"github.com/Originate/exosphere/exo-go/src/types"
+	"github.com/Originate/exosphere/exo-go/src/util"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("GetServiceDockerConfigs", func() {
+
+	var setup *appSetup.AppSetup
+	var dockerCompose types.DockerCompose
+	var appDir string
+	var internalServices, externalServices, internalDependencies, externalDependencies, allServices []string
+
+	var _ = BeforeSuite(func() {
+		err := appSetup.CheckoutApp(os.TempDir(), "complex-setup-app")
+		Expect(err).NotTo(HaveOccurred())
+		appDir = path.Join(os.TempDir(), "complex-setup-app")
+		appConfig, err := appConfigHelpers.GetAppConfig(appDir)
+		Expect(err).NotTo(HaveOccurred())
+		setup, err = appSetup.NewAppSetup(appConfig, logger.NewLogger([]string{}, []string{}), appDir)
+		Expect(err).NotTo(HaveOccurred())
+
+		internalServices = []string{"html-server", "todo-service", "users-service"}
+		externalServices = []string{"external-service"}
+		internalDependencies = []string{"exocom0.22.1"}
+		externalDependencies = []string{"mongo3.4.0"}
+		allServices = util.JoinSlices(internalServices, externalServices, internalDependencies, externalDependencies)
+	})
+
+	var _ = Describe("StartSetup", func() {
+
+		It("should starts up the app sucessfully when application.yml and service.yml are valid", func() {
+			err := setup.StartSetup()
```

Comment:
Please update the tests to not rely on each other. Put anything you need in each test in a beforeEach / beforeSuite. If there are some things that only needed in a few of the tests wrap them in a describe block so it can have its own beforeEach